aspnetcore/security/blazor/server-side.md

```diff
@@ -0,0 +1,378 @@
+  * Cause an issue in the app by returning an error from the JavaScript function.
+  * Induce an unintended behavior on the server by returning an unexpected result from the JavaScript function.
+
+<!-- I refactored "our clients" into "Blazor app client running in the browser" ... did I get that right? ... rephrase? -->
+
+Take the following precautions to guard against the preceding scenarios:
+
+* Wrap JS interop calls within [try-catch](/dotnet/csharp/language-reference/keywords/try-catch) statements to account for errors that might occur during the invocations. For more information, see <xref:blazor/handle-errors#javascript-interop>.
+* Validate data returned from JS interop invocations, including error messages, before taking any action.
+
+### .NET methods invoked from the browser
+
+Don't trust calls from JavaScript to .NET methods. When a .NET method is exposed to JavaScript, consider how the .NET method is invoked:
+
+* Treat any .NET method exposed to JavaScript as you would a public endpoint to the app.
+  * Validate input.
+    * Ensure that values are within expected ranges.
+    * Ensure that the user has permission to perform the action requested.
+  * Don't allocate an excessive quantity of resources as part of the .NET method invocation. For example, perform checks and place limits on CPU and memory use.
+  * Take into account that static and instance methods can be exposed to JavaScript clients. Avoid sharing state across sessions unless the design calls for sharing state with appropriate constraints.
+    * For instance methods exposed through `DotNetReference` objects that are originally created through dependency injection (DI), the objects should be registered as scoped objects. This applies to any DI service that the Blazor server-side app uses.
+    * For static methods, avoid establishing state that can't be scoped to the client unless the app is explicitly sharing state by-design across all users on a server instance.
+  * Avoid passing user-supplied data in parameters to JavaScript calls. If passing data in parameters is absolutely required, ensure that the JavaScript code handles passing the data without introducing [Cross-site scripting (XSS)](#cross-site-scripting-xss) vulnerabilities. For example, don't write user-supplied data to the Document Object Model (DOM) by setting the `innerHTML` property of an element. Consider using [Content Security Policy (CSP)](https://developer.mozilla.org/docs/Web/HTTP/CSP) to disable `eval` and other unsafe JavaScript primitives.
+* Avoid implementing custom dispatching of .NET invocations on top of the framework's dispatching implementation. Exposing .NET methods to the browser is an advanced scenario, not recommended for general Blazor development.
```

Comment:
I don't really get what this means and what we are trying to protect against here. @javiercn?

Reply:
You can create a JSInvokable .NET method that takes in a string and use that string to call random .NET methods if you want to bypass the JSInvokable attribute. This is telling you to avoid doing that